src/extensions/default/JavaScriptCodeHints/ParameterHintsProvider.js

```diff
@@ -0,0 +1,229 @@
+    };
+
+    /**
+     * Save the state of the current hint. Called when popping up a parameter hint
+     * for a parameter, when the parameter already part of an existing parameter
+     * hint.
+     */
+    JSParameterHintsProvider.prototype.pushHintOnStack = function () {
+        this.hintStack.push(this.hintState);
+    };
+
+    /**
+     * Restore the state of the previous function hint.
+     *
+     * @return {boolean} - true the a parameter hint has been popped, false otherwise.
+     */
+    JSParameterHintsProvider.prototype.popHintFromStack = function () {
+        if (this.hintStack.length > 0) {
+            this.hintState = this.hintStack.pop();
+            this.hintState.visible = false;
+            return true;
+        }
+
+        return false;
+    };
+
+    /**
+     * Reset the function hint stack.
+     */
+    JSParameterHintsProvider.prototype.clearFunctionHintStack = function () {
+        this.hintStack = [];
+    };
+
+    /**
+     * Test if the function call at the cursor is different from the currently displayed
+     * function hint.
+     *
+     * @param {{line:number, ch:number}} functionCallPos - the offset of the function call.
+     * @return {boolean}
+     */
+    JSParameterHintsProvider.prototype.hasFunctionCallPosChanged = function (functionCallPos) {
+        var oldFunctionCallPos = this.hintState.functionCallPos;
+        return (oldFunctionCallPos === undefined ||
+            oldFunctionCallPos.line !== functionCallPos.line ||
+            oldFunctionCallPos.ch !== functionCallPos.ch);
+    };
+
+    /**
+     * Dismiss the function hint.
+     *
+     */
+    JSParameterHintsProvider.prototype.cleanHintState = function () {
+        if (this.hintState.visible) {
+            if (!this.preserveHintStack) {
+                this.clearFunctionHintStack();
+            }
+        }
+    };
+
+    /**
+     * Pop up a function hint on the line above the caret position.
+     *
+     * @param {boolean=} pushExistingHint - if true, push the existing hint on the stack. Default is false, not
+     * to push the hint.
+     * @param {string=} hint - function hint string from tern.
+     * @param {{inFunctionCall: boolean, functionCallPos:
+     * {line: number, ch: number}}=} functionInfo -
+     * if the functionInfo is already known, it can be passed in to avoid
+     * figuring it out again.
+     * @return {jQuery.Promise} - The promise will not complete until the
+     *      hint has completed. Returns null, if the function hint is already
+     *      displayed or there is no function hint at the cursor.
+     *
+     */
+    JSParameterHintsProvider.prototype._getParameterHint = function (pushExistingHint, hint, functionInfo) {
+        var result = $.Deferred();
+        functionInfo = functionInfo || this.session.getFunctionInfo();
+        if (!functionInfo.inFunctionCall) {
+            this.cleanHintState();
+            result.reject(null);
```

Comment:
Fixed. Thanks for the catch. It can return early. Basically, since we were returning a promise which could be resolved only once, it was still functioning correctly.